probe.xs

```diff
@@ -65,41 +71,67 @@ static inline void probe_invoke_callback(const char* file, int line, SV* callbac
     LEAVE;
 }
 
-static int probe_lookup(const char* file, int line, int action)
+static bool probe_lookup(const char* file, int line, int action)
 {
     U32 klen = strlen(file);
     char kstr[20];
     SV** rlines = 0;
-    SV** rflag = 0;
     HV* lines = 0;
-    SV* flag = 0;
 
     rlines = hv_fetch(probe_hash, file, klen, 0);
     if (rlines) {
         lines = (HV*) SvRV(*rlines);
         TRACE(("PROBE found entry for file [%s]: %p\n", file, lines));
     } else {
-        return PROBE_TYPE_NONE;
+        return false;
     }
 
     klen = sprintf(kstr, "%d", line);
-    rflag = hv_fetch(lines, kstr, klen, 0);
-    if (rflag) {
-        int ret = 0;
-        flag = *rflag;
-        ret = SvIV(flag);
+    if (hv_exists(lines, kstr, klen)) {
         if (action == PROBE_ACTION_REMOVE) {
             /* TODO: remove file name when last line for file was removed? */
             hv_delete(lines, kstr, klen, G_DISCARD);
-            INFO(("PROBE removed entry for line [%s] => %d\n", kstr, ret));
+            INFO(("PROBE removed entry for line [%s]\n", kstr));
         }
-        return ret;
+        return true;
     } else {
-        return PROBE_TYPE_NONE;
+        return false;
     }
 
     /* catch any mistakes */
-    return PROBE_TYPE_NONE;
+    return false;
+}
+
+static AV* probe_settings(const char* file, int line)
+{
+    U32 klen = strlen(file);
+    char kstr[20];
+    SV** rlines = 0;
+    HV* lines = 0;
+    SV** rsettings = 0;
+    AV* settings = 0;
+
+    rlines = hv_fetch(probe_hash, file, klen, 0);
+    if (rlines) {
+        lines = (HV*) SvRV(*rlines);
+    } else {
+        return 0;
+    }
+
```

Comment:
I think it is clearer to do early returns here:
```C
if (!rlines) {
    return 0;
}
lines = (HV*) SvRV(*rlines);
```

Reply:
Good call, thanks